Library/Homebrew/dependency.rb

```diff
@@ -55,14 +56,32 @@ def installed?(minimum_version: nil)
     return false unless formula
 
     if minimum_version.present?
-      formula.any_version_installed? && (formula.any_installed_version.version >= minimum_version)
+      installed_version = formula.any_installed_version
+      return false unless installed_version
+
+      # Tabs prior to 4.1.18 did not have revision or pkg_version fields.
+      # As a result, we have to be more conversative when we do not have
+      # a minimum revision from the tab and assume that if the formula has a
+      # the same version and a non-zero revision that it needs upgraded.
+      if minimum_revision.present?
+        minimum_pkg_version = PkgVersion.new(minimum_version, minimum_revision)
+        installed_version >= minimum_pkg_version
+      else
+        minimum_pkg_version = PkgVersion.new(minimum_version, 0)
+        if installed_version == minimum_pkg_version
```

Comment:
```suggestion
        if installed_version.version == minimum_pkg_version.version
```

Reply:
Can probably ditch the `PkgVersion.new` to make it more obvious - I'll push that now.